src/backend/base/langflow/components/data/rss.py

```
@@ -0,0 +1,53 @@
+import pandas as pd
+import requests
+from bs4 import BeautifulSoup
+
+from langflow.custom import Component
+from langflow.io import MessageTextInput, Output
+from langflow.logging import logger
+from langflow.schema import DataFrame
+
+
+class RSSReaderComponent(Component):
+    display_name = "RSS Reader"
+    description = "Fetches and parses an RSS feed."
+    icon = "rss"
+    name = "RSSReaderSimple"
+
+    inputs = [
+        MessageTextInput(
+            name="rss_url",
+            display_name="RSS Feed URL",
+            info="URL of the RSS feed to parse.",
+            tool_mode=True,
+            input_types=[],
+            required=True,
+        )
+    ]
+
+    outputs = [Output(name="articles", display_name="Articles", method="read_rss")]
+
+    def read_rss(self) -> DataFrame:
+        try:
+            response = requests.get(self.rss_url, timeout=5)
+            response.raise_for_status()
+            soup = BeautifulSoup(response.content, "xml")
+            items = soup.find_all("item")
+        except (requests.RequestException, ValueError) as e:
+            self.status = f"Failed to fetch RSS: {e}"
+            return DataFrame(pd.DataFrame([{"title": "Error", "link": "", "published": "", "summary": str(e)}]))
```

Do we need to return a DataFrame instead of error handling here?

---

We can raise an error instead.
Good Catch, Can you test it with tool mode also?